bundle/src/main/java/com/adobe/acs/commons/packagegarbagecollector/PackageGarbageCollectionJob.java

```diff
@@ -0,0 +1,126 @@
+
+import org.apache.jackrabbit.vault.packaging.JcrPackage;
+import org.apache.jackrabbit.vault.packaging.JcrPackageManager;
+import org.apache.jackrabbit.vault.packaging.Packaging;
+import org.apache.sling.api.resource.LoginException;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.apache.sling.api.resource.ResourceResolverFactory;
+import org.apache.sling.event.jobs.Job;
+import org.apache.sling.event.jobs.consumer.JobConsumer;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.jcr.Node;
+import javax.jcr.RepositoryException;
+import javax.jcr.Session;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.Period;
+import java.time.format.DateTimeFormatter;
+import java.time.format.FormatStyle;
+import java.util.Calendar;
+import java.util.Collections;
+import java.util.List;
+
+import static com.adobe.acs.commons.packagegarbagecollector.PackageGarbageCollectionScheduler.*;
+
+@Component(
+    service = JobConsumer.class,
+    immediate = true,
+    property = { JobConsumer.PROPERTY_TOPICS + "=" + PackageGarbageCollectionScheduler.JOB_TOPIC })
+public class PackageGarbageCollectionJob implements JobConsumer {
+    private static final Logger LOG = LoggerFactory.getLogger(PackageGarbageCollectionJob.class);
+
+    private static final String SERVICE_USER = "package-garbage-collection";
+
+    @Reference
+    Packaging packaging;
+
+    @Reference
+    ResourceResolverFactory resourceResolverFactory;
+
+    @Override
+    public JobResult process(Job job) {
+        Session session = null;
+        String groupName = job.getProperty(GROUP_NAME, String.class);
+        Integer maxAgeInDays = job.getProperty(MAX_AGE_IN_DAYS, Integer.class);
+        LOG.debug("Job Configuration: ["
+                + "Group Name: {}, "
+                + "Service User: {}, "
+                + "Age of Package {} days,]", groupName, SERVICE_USER, maxAgeInDays);
+
+        try (ResourceResolver resourceResolver = resourceResolverFactory.getServiceResourceResolver(
+            Collections.singletonMap(ResourceResolverFactory.SUBSERVICE, SERVICE_USER))) {
+            session = resourceResolver.adaptTo(Session.class);
+            JcrPackageManager packageManager = packaging.getPackageManager(session);
+            List<JcrPackage> packages = packageManager.listPackages(groupName, false);
+
+            for (JcrPackage jcrPackage : packages) {
+                String packageDescription = getPackageDescription(jcrPackage);
+                if (isPackageOldEnough(jcrPackage, maxAgeInDays)) {
+                    packageManager.remove(jcrPackage);
+                    LOG.info("Deleted package {}", packageDescription);
+                } else {
+                    LOG.info("Not removing package because it's not old enough {}", packageDescription);
```

Comment:
I would downgrade this to DEBUG. Don't log no-actions.

```suggestion
                    LOG.debug("Not removing package because it's not old enough {}", packageDescription);
```